forte/processors/base/data_augment_processor.py

```diff
@@ -27,8 +28,11 @@
 
 class BaseDataAugmentProcessor(Caster[DataPack, MultiPack]):
     r"""The base class of processors that augments the data.
+    Pass the input data_pack to the "cast" method
+    and it will return an output Multipack.
+
+    To inherit from this class, please override the augment function.
```

Comment:
The docstring here can still be improved. 

1. Users don't have to pass in data pack to `cast` themselves.
2. To refer a function in the docstring, see here https://stackoverflow.com/questions/11168178/how-do-i-reference-a-documented-python-function-parameter-using-sphinx-markup

The docstring here should explain what this processor do. "Augments the data" is too vague.  "return an output Multipack" is also not informational